File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hMap = new HashMap<>();
        int[] result = new int[2];
        for (int i = 0; i < nums.length; i++) {
            int current = nums[i];
            int diff = target - current;
            if (hMap.containsKey(diff)) {
                result[0] = hMap.get(diff);
                result[1] = i;
            }
            hMap.put(current, i);
        }
        return result;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l, r = 0, 1
        maxProfit = 0

        while r < len(prices):
            if prices[l] < prices[r]:
                diff = prices[r] - prices[l]
                maxProfit = max(diff, maxProfit)
            else:
                l = r
            r += 1

        return maxProfit

        
